--- a/src/open_mower/launch/gamepad_remote_control.launch
+++ b/src/open_mower/launch/gamepad_remote_control.launch
@@ -7,5 +7,5 @@
     <include file="$(find open_mower)/launch/include/_localization.launch"/>
     <include file="$(find open_mower)/launch/include/_teleop.launch"/>
 
-    <node pkg="mower_map" type="mower_map_service" name="map_service" output="screen"/>
+    <node pkg="mower_map" type="mower_map_service" name="map_service"/>
 </launch>
--- a/src/open_mower/launch/include/_localization.launch
+++ b/src/open_mower/launch/include/_localization.launch
@@ -1,5 +1,5 @@
 <launch>
-    <node pkg="xbot_positioning" type="xbot_positioning" name="xbot_positioning" output="screen" required="true">
+    <node pkg="xbot_positioning" type="xbot_positioning" name="xbot_positioning" required="true">
         <remap from="~imu_in" to="/ll/imu/data_raw"/>
         <remap from="~twist_in" to="/ll/diff_drive/measured_twist"/>
         <remap from="~xb_pose_in" to="/ll/position/gps"/>
--- a/src/open_mower/launch/include/_move_base.launch
+++ b/src/open_mower/launch/include/_move_base.launch
@@ -1,6 +1,6 @@
 <launch>
 
-  <node name="move_base_flex" pkg="mbf_costmap_nav" type="mbf_costmap_nav" required="true" output="screen">
+  <node name="move_base_flex" pkg="mbf_costmap_nav" type="mbf_costmap_nav" required="true">
     <remap from="cmd_vel" to="/nav_vel"/>
     <rosparam file="$(find open_mower)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
     <rosparam file="$(find open_mower)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
--- a/src/open_mower/launch/open_mower.launch
+++ b/src/open_mower/launch/open_mower.launch
@@ -9,29 +9,29 @@
     </include>
 
 
-    <node pkg="mower_map" type="mower_map_service" name="map_service" output="screen"/>
-    <node pkg="mower_logic" type="mower_logic" name="mower_logic" output="screen">
+    <node pkg="mower_map" type="mower_map_service" name="map_service"/>
+    <node pkg="mower_logic" type="mower_logic" name="mower_logic">
     </node>
-    <node pkg="slic3r_coverage_planner" type="slic3r_coverage_planner" name="slic3r_coverage_planner" output="screen"/>
+    <node pkg="slic3r_coverage_planner" type="slic3r_coverage_planner" name="slic3r_coverage_planner"/>
 
-    <node pkg="twist_mux" type="twist_mux" name="twist_mux" output="screen">
+    <node pkg="twist_mux" type="twist_mux" name="twist_mux">
         <remap from="cmd_vel_out" to="/ll/cmd_vel"/>
 
         <rosparam file="$(find open_mower)/params/twist_mux_topics.yaml" command="load"/>
     </node>
 
-    <node pkg="xbot_monitoring" type="xbot_monitoring" name="xbot_monitoring" output="screen" respawn="true"
+    <node pkg="xbot_monitoring" type="xbot_monitoring" name="xbot_monitoring" respawn="true"
           respawn_delay="10">
         <remap from="/xbot_monitoring/remote_cmd_vel" to="/joy_vel"/>
     </node>
 
-    <node pkg="xbot_monitoring" type="heatmap_generator" name="heatmap_generator" output="screen" respawn="true"
+    <node pkg="xbot_monitoring" type="heatmap_generator" name="heatmap_generator" respawn="true"
           respawn_delay="10" unless="$(eval optenv('OM_HEATMAP_SENSOR_IDS', 'UNSET') == 'UNSET')">
     </node>
 
-    <node pkg="xbot_remote" type="xbot_remote" name="xbot_remote" output="screen" respawn="true" respawn_delay="10">
+    <node pkg="xbot_remote" type="xbot_remote" name="xbot_remote" respawn="true" respawn_delay="10">
         <remap from="/xbot_remote/cmd_vel" to="/joy_vel"/>
     </node>
 
-    <node pkg="mower_logic" type="monitoring" name="monitoring" output="screen" respawn="true" respawn_delay="10"/>
+    <node pkg="mower_logic" type="monitoring" name="monitoring" respawn="true" respawn_delay="10"/>
 </launch>
--- a/src/open_mower/launch/sim_navigation.launch
+++ b/src/open_mower/launch/sim_navigation.launch
@@ -11,7 +11,7 @@
           required="true"/>
     <node pkg="mower_map" type="mower_map_service" name="mower_map" required="true"/>
 
-    <node pkg="twist_mux" type="twist_mux" name="twist_mux" output="screen">
+    <node pkg="twist_mux" type="twist_mux" name="twist_mux">
         <remap from="cmd_vel_out" to="/cmd_vel"/>
 
         <rosparam file="$(find open_mower)/params/twist_mux_topics.yaml" command="load"/>
