--- a/src/lib/slic3r_coverage_planner/CMakeLists.txt
+++ b/src/lib/slic3r_coverage_planner/CMakeLists.txt
@@ -36,7 +36,7 @@ ExternalProject_Add(
         CMAKE_CACHE_ARGS
         SOURCE_SUBDIR src
         CMAKE_ARGS -DSLIC3R_BUILD_TESTS=OFF
-        BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release --parallel ${NPROC}
+        BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release --parallel 1
 )
 
 
