Index: open_mower_ros/src/lib/slic3r_coverage_planner/CMakeLists.txt
===================================================================
--- open_mower_ros.orig/src/lib/slic3r_coverage_planner/CMakeLists.txt
+++ open_mower_ros/src/lib/slic3r_coverage_planner/CMakeLists.txt
@@ -37,7 +37,7 @@ ExternalProject_Add(
         CMAKE_CACHE_ARGS
         SOURCE_SUBDIR src
         CMAKE_ARGS -DSLIC3R_BUILD_TESTS=OFF
-        BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release --parallel ${NPROC}
+        BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release --parallel 1
 )
 
 
